src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -37,7 +37,10 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
         val blockTimestamp = Instant.now.getEpochSecond
         val header: BlockHeader = prepareHeader(blockNumber, ommers, beneficiary, parent, blockTimestamp)
 
-        val transactionsForBlock = transactions.toSet[SignedTransaction].toList.sortBy(_.tx.nonce)
+        val transactionsForBlock = transactions
+          .filter(_.tx.gasLimit < header.gasLimit)
+          //if we have 2 transactions from same address we want first one with lower nonce
+          .sortBy(_.tx.nonce)
```

Comment:
I'm not saying this is wrong at all but I would imagine that also using the fee (indirectly by gasPrice) should be used to pick tx to mine first.

This is just a commet, this shouldn't be a stopper for merging.